sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c

```diff
@@ -174,33 +174,37 @@ int gnrc_pktbuf_realloc_data(gnrc_pktsnip_t *pkt, size_t size)
                           _align(sizeof(_unused_t)) : _align(size);
 
     mutex_lock(&_mutex);
-    assert((pkt != NULL) && (pkt->data != NULL) && _pktbuf_contains(pkt->data));
-    if (size == 0) {
-        DEBUG("pktbuf: size == 0\n");
-        mutex_unlock(&_mutex);
-        return ENOMEM;
-    }
+    assert((pkt != NULL) && ((pkt->data == NULL) || _pktbuf_contains(pkt->data)));
```

Comment:
if `pkt->data` is NULL, then `pkt->size` must also be `0`, or am I mistaken? If that's the case, then both should be checked with `&&`. Maybe the statement is getting too complicated and we should split that apart into multiple assert statements?